tests/solvers/linear/univariate_test.py

```diff
@@ -163,3 +168,31 @@ def univ_dist(x, y, a, b):
     expected = univ_dist(x, y, a, b + v_b) - univ_dist(x, y, a, b - v_b)
     actual = 2.0 * jnp.vdot(v_b, grad_b)
     np.testing.assert_allclose(actual, expected, rtol=tol, atol=tol)
+
+  @pytest.mark.fast()
+  def test_dual_vectors(self):
+    """The OTT solver outputs meaningful dual variables."""
+    _ = pytest.importorskip("lineax")  # only tested using lineax
+    n = self.n
+    x, a, y, b, z, c = self.x, self.a, self.y, self.b, self.z, self.c
+    unif_n = jnp.ones((n,)) / n
+    for (target, weights_target) in ((y, b), (z, c)):
```

Comment:
In general, am not a fan of this sort of tests; one way to solve this is to `pytest.mark.parametrize(("target", "weights"), [("y", "b"), ...])`
and use `target = getattr(self, target)` to get the correct array in the code.